### PR TITLE
[API] Introducing remote debugging

### PR DESCRIPTION
**Pull request by liranbg**
Instructions is written on `CONTRIBUTING.md` file

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -115,6 +115,22 @@ jobs:
             make push-${{ matrix.image-name }}; \
           done;
 
+    - name: Pull cache, build and push dev image
+      if: ${{ matrix.image-name == 'api' }}
```
Comment (Tankilevitch) [inline .github/workflows/build.yaml:119]: We have a mechanism for skipping images, let's add an option to skip `api-dev` as well.

File: Makefile [hunk 2]
```diff
@@ -304,13 +304,16 @@ push-jupyter: jupyter ## Push mlrun jupyter docker image
 
 
 MLRUN_API_IMAGE_NAME := $(MLRUN_DOCKER_IMAGE_PREFIX)/mlrun-api
+MLRUN_API_DEV_IMAGE_NAME := $(MLRUN_DOCKER_IMAGE_PREFIX)/mlrun-api-dev
 MLRUN_API_CACHE_IMAGE_NAME := $(MLRUN_CACHE_DOCKER_IMAGE_PREFIX)/mlrun-api
 MLRUN_API_IMAGE_NAME_TAGGED := $(MLRUN_API_IMAGE_NAME):$(MLRUN_DOCKER_TAG)
+MLRUN_API_DEV_IMAGE_NAME_TAGGED := $(MLRUN_API_IMAGE_NAME):$(MLRUN_DOCKER_TAG)-dev
```
Comment (Tankilevitch) [inline Makefile:310]:
```suggestion
MLRUN_API_DEV_IMAGE_NAME_TAGGED := $(MLRUN_API_IMAGE_NAME)-dev:$(MLRUN_DOCKER_TAG)
```

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
This is so useful!! Nicely done, added minor comments

**Comment by liranbg** [comment]
continue against development branch @ https://github.com/mlrun/mlrun/pull/2265

**Pull request closed without merging**